### PR TITLE
fix: [SRM-11705]: multiple bug fixes

### PR DESCRIPTION
**Pull request by deepesh-ui**
### Summary

- https://harness.atlassian.net/browse/SRM-11705
- https://harness.atlassian.net/browse/SRM-11753
- https://harness.atlassian.net/browse/SRM-11754
- https://harness.atlassian.net/browse/SRM-11779
- https://harness.atlassian.net/browse/SRM-11798
- https://harness.atlassian.net/browse/SRM-11800

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11069)

File: src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJsonMultiType.tsx [hunk 1]
```diff
@@ -37,7 +37,8 @@ export const InputWithDynamicModalForJsonMultiType = (
             if (isMultiTypeRuntime(valueType) || valueType === MultiTypeInputType.EXPRESSION) {
               props.onChange(props.inputName as string, key as string)
             } else if (valueType === MultiTypeInputType.FIXED) {
-              props.onChange(props.inputName as string, value as string)
+              const fixedValue = key ? value : ''
+              props.onChange(props.inputName as string, fixedValue as string)
```
Comment (harshilbits) [inline src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJsonMultiType.tsx:41]: destructure fields from props.

File: cypress/integration/85-cv/HealthSource/AppDynamics.spec.ts [hunk 2]
```diff
@@ -127,9 +127,9 @@ describe('Create empty monitored service', () => {
     cy.get('input[name="Errors"]').uncheck({ force: true })
     cy.get('input[name="Performance"]').uncheck({ force: true })
     cy.contains('span', 'Submit').click({ force: true })
-    cy.contains('span', 'Plese select metric packs').should('be.visible')
+    cy.contains('span', 'Please select atleast one metric pack').should('be.visible')
```
Comment (harshilbits) [inline cypress/integration/85-cv/HealthSource/AppDynamics.spec.ts:130]: String is 'Please select atleast one metric pack' is repeated a lot. Please move it to constants and reuse.

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJsonMultiType.tsx (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJsonMultiType.tsx:41]: done

File: cypress/integration/85-cv/HealthSource/AppDynamics.spec.ts (hunk 2 above)
Comment (deepesh-ui) [reply cypress/integration/85-cv/HealthSource/AppDynamics.spec.ts:130]: adding a ticket to move all validation text to constant

Comment (deepesh-ui) [reply cypress/integration/85-cv/HealthSource/AppDynamics.spec.ts:130]: https://harness.atlassian.net/browse/SRM-11923

**Review by praneshg239 — APPROVED** [review]

File: cypress/integration/85-cv/HealthSource/AppDynamics.spec.ts (hunk 2 above)
Comment (harshilbits) [reply cypress/integration/85-cv/HealthSource/AppDynamics.spec.ts:130]: This should not take much time...cant we do in this PR ?

Comment (harshilbits) [reply cypress/integration/85-cv/HealthSource/AppDynamics.spec.ts:130]: As discussed , please mention all the common texts to be moved in constants in a ticket and pick it up on priority.

**Review by harshilbits — APPROVED** [review]

**Pull request merged**